src/main/java/com/yelp/nrtsearch/server/luceneserver/doc/VectorType.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.doc;
+
+import java.lang.reflect.Array;
+import java.util.AbstractList;
+
+public final class VectorType extends AbstractList<Float> {
+
+  private static final int DEFAULT_CAPACITY = 10;
+
+  private float[] vectorData;
+
+  private int vectorSize;
+
+  public VectorType() {
+    this(DEFAULT_CAPACITY);
+  }
+
+  /**
+   * Construct VectorType with given vector data
+   *
+   * @param vectorData
+   */
+  public VectorType(float[] vectorData) {
+    this.vectorData = vectorData;
+    this.vectorSize = vectorData.length;
+  }
+
+  /**
+   * Constructor to create a float array with given initial capacity
+   *
+   * @param initialCapacity initial vector capacity @Throws IllegalArgumentException – if the
+   *     specified capacity is negative
+   */
+  public VectorType(int initialCapacity) {
+    super();
+    if (initialCapacity < 0) {
+      throw new IllegalArgumentException("vector capacity should be >= 0");
+    }
+    this.vectorData = new float[initialCapacity];
+  }
```

Comment:
If a constructor for an empty vector is provided, the `set` method should probably be overridden and implemented.